cmd/hatchet-engine/engine/run.go

```diff
@@ -237,8 +249,18 @@ func Run(ctx context.Context, cf *loader.ConfigLoader) error {
 
 	l.Debug().Msgf("engine has started")
 
+	if healthProbes {
+		h.SetReady(true)
+	}
+
 	<-ctx.Done()
 
+	if healthProbes {
+		h.SetReady(false)
+	}
+
+	time.Sleep(10 * time.Second)
```

Comment:
This is not needed, however I was thinking that there is most likely a delay which is at least a few seconds until any orchestrator (k8s or compose) registers this. So I figured it would be best to set the probe false first, give the orchestrator some time to register, and only then start with the actual shutdown of services.

Reply:
Yes, we can set the interval of the health check on the k8s side, and we should indeed wait for that interval times the number of retries (in the case that the kube-proxy cannot reach the pod for the number of retries). So if the interval is 5 seconds and retries is 3, we should sleep for 15 seconds. Because we'll probably be changing this value, can we make this a configurable env var?

Reply:
Makes sense! Done.